### PR TITLE
Added "make debug" which builds with DEBUG_F flag

### PR DESCRIPTION
**Pull request by hoodmane**
DEBUG_F provides extra logging info: whenever `FAIL` is invoked, a messages is printed with a javascript stack trace, and the source line function and file where the failure happened. When #1080 is merged, it will add debug statements whenever an error occurs in `EM_JS`.

I also added a missing semicolon in the `DEBUG_F` version of `FAIL`.

### CODE REVIEW

**Comment by dalcde** [comment]
Should debug enable `-s ASSERTIONS=2` as well?

**Comment by hoodmane** [comment]
Let me check how much difference that makes to the linking time.

**Comment by hoodmane** [comment]
Okay they take the same amount of time so I'll add it.

**Comment by hoodmane** [comment]
Btw, with `-s ASSERTIONS=2` there's a bunch of weird warnings like:
`warning: unexpected number of arguments 0 in call to 'hiwire_array', should be 1`

Also, no matter what it always says:
`emcc:INFO: Enabling --no-heap-copy because -s ALLOW_MEMORY_GROWTH=1 is being used with file_packager.py (pass --no-heap-copy to suppress this notification)`

**Comment by hoodmane** [comment]
Also instead of `PYODIDE_PACKAGES="micropip"` I should probably make it do all the packages for the core tests.

**Comment by hoodmane** [comment]
Oh this now has the problem that it's annoying to get an extra package built because the recipe writes over `PYODIDE_PACKAGES`. Maybe `PYODIDE_PACKAGES+=...`?

**Comment by hoodmane** [comment]
@dalcde @rth I think this is a simple enough addition, would be helpful to merge (I will switch from using `make minimal` to `make debug` as my default).

File: Makefile [hunk 1]
```diff
@@ -203,3 +205,8 @@ check:
 
 minimal :
 	PYODIDE_PACKAGES="micropip" make
+
+debug :
+	EXTRA_CFLAGS="-D DEBUG_F -s ASSERTIONS=2" \
```
Comment (dalcde) [inline Makefile:210]: I think `ASSERTIONS=2` should be an ldflag. It affects the js code gen

Comment (hoodmane) [reply Makefile]: Funny, I just noticed that as I was working on #1126.

**Pull request merged**